src/thunderbots/software/test/world/team.cpp

```diff
@@ -0,0 +1,396 @@
+    team.updateRobots({}, two_seconds_100ms_future);
+
+    EXPECT_EQ(0, team.size());
+    EXPECT_EQ(std::nullopt, team.getRobotById(0));
+    EXPECT_EQ(std::nullopt, team.getRobotById(1));
+}
+
+TEST_F(TeamTest, update_robots_staggered_robot_expiry)
+{
+    Team team = Team(milliseconds(2500));
+
+    Robot robot_0 = Robot(0, Point(0, 1), Vector(-1, -2), Angle::half(),
+                          AngularVelocity::threeQuarter(), current_time);
+
+    team.updateRobots({robot_0}, current_time);
+
+    EXPECT_EQ(1, team.size());
+    EXPECT_EQ(robot_0, team.getRobotById(0));
+    EXPECT_EQ(std::nullopt, team.getRobotById(1));
+    EXPECT_EQ(std::nullopt, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+
+    // Add a robot 1 second later. The first robot has not expired yet
+    Robot robot_1 = Robot(1, Point(3, -1), Vector(), Angle::zero(),
+                          AngularVelocity::zero(), one_second_future);
+
+    team.updateRobots({robot_1}, one_second_future);
+
+    EXPECT_EQ(2, team.size());
+    EXPECT_EQ(robot_0, team.getRobotById(0));
+    EXPECT_EQ(robot_1, team.getRobotById(1));
+    EXPECT_EQ(std::nullopt, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+
+    // Add a robot another second later. The first 2 robots have not expired yet
+
+    Robot robot_2 = Robot(2, Point(), Vector(-0.5, 4), Angle::quarter(),
+                          AngularVelocity::half(), two_seconds_future);
+
+    team.updateRobots({robot_2}, two_seconds_future);
+
+    EXPECT_EQ(3, team.size());
+    EXPECT_EQ(robot_0, team.getRobotById(0));
+    EXPECT_EQ(robot_1, team.getRobotById(1));
+    EXPECT_EQ(robot_2, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+
+    // Update with no new data. It has now been 3 seconds since the first robot was added,
+    // which is past the debounce threshold. Therefore we expect the first robot to have
+    // been removed from the team
+    team.updateRobots({}, three_seconds_future);
+
+    EXPECT_EQ(2, team.size());
+    EXPECT_EQ(std::nullopt, team.getRobotById(0));
+    EXPECT_EQ(robot_1, team.getRobotById(1));
+    EXPECT_EQ(robot_2, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+
+    // Update again. The second robot should have expired now
+    team.updateRobots({}, four_seconds_future);
+
+    EXPECT_EQ(1, team.size());
+    EXPECT_EQ(std::nullopt, team.getRobotById(0));
+    EXPECT_EQ(std::nullopt, team.getRobotById(1));
+    EXPECT_EQ(robot_2, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+
+    // Update again. The third robot should have expired now
+    team.updateRobots({}, five_seconds_future);
+
+    EXPECT_EQ(0, team.size());
+    EXPECT_EQ(std::nullopt, team.getRobotById(0));
+    EXPECT_EQ(std::nullopt, team.getRobotById(1));
+    EXPECT_EQ(std::nullopt, team.getRobotById(2));
+    EXPECT_EQ(std::nullopt, team.getRobotById(3));
+}
```

Comment:
This feels more like an integration test then a unit test. We're covering a _lot_ of cases, any way we can break this up?

Reply:
done